src/backend/crud/deployment.py

```diff
@@ -184,3 +190,33 @@ def delete_deployment(db: Session, deployment_id: str) -> None:
     deployment = db.query(Deployment).filter(Deployment.id == deployment_id)
     deployment.delete()
     db.commit()
+
+
+@validate_transaction
+def create_deployment_by_config(db: Session, deployment_config: DeploymentSchema) -> Deployment:
+    """
+    Create a new deployment by config.
+
+    Args:
+        db (Session): Database session.
+        deployment (str): Deployment data to be created.
+        deployment_config (DeploymentSchema): Deployment config.
+
+    Returns:
+        Deployment: Created deployment.
+    """
+    deployment = Deployment(
+        name=deployment_config.name,
+        description="",
+        default_deployment_config= {
+                env_var: os.environ.get(env_var, "")
```

Comment:
This would create a Deployment when an env variable could be unset with the `""` default, open to how to handle that scenario. Maybe just iterate through the variables before creating the deployment and check they're not None

Reply:
I don't think we need to do this in the agent creation flow. I believe this check should be part of the Chat flow, and it’s already covered by the models. What do you think?